File: Ruby/hw4/utVector.rb
require ('test/unit')
require ('./Vector')

class TestVector < Test::Unit::TestCase
    def testConstructor
        v = Vector.new([1, 2])
        assert_equal([1, 2], v.component)
        assert_equal(2, v.dimension)
        assert_equal(1, v.component(1))
        assert_equal(2, v.component(2))
    end
    def testConstructorException
        assert_raise_message("Dimension Error", ArgumentError) do
            Vector.new([])
        end
    end

    def testLength
        v = Vector.new([3, 4])
        assert_equal(5, v.length)
    end

    def testGetVector
        s = "(1, 2)"
        assert_equal([1, 2], getVector(s).component)
    end
end

File: JavaScript/hw4/utVector.js
describe('Test Vector', function(){
    it('Constructor', function(){
        var v = new Vector([1, 2]);
        expect(2).toEqual(v.dimension());
        expect([1, 2]).toEqual(v.component());
        expect(1).toEqual(v.component({'index':1}));
        expect(2).toEqual(v.component({'index':2}));
    });
});

describe('Test Vector Length', function(){
    it('Normal', function(){
        var v = new Vector([3, 4]);
        expect(5).toEqual(v.length());
    });
});

describe('Test getVector', function(){
    it('Normal', function(){
        var s = '(1,2)';
        var v = getVector(s);
        expect(2).toEqual(v.dimension());
        expect(1).toEqual(v.component({'index':1}));
        expect(2).toEqual(v.component({'index':2}));
    });
    it('Exception', function(){
        var s = '1,2))';
        var error = function(){
            getVector(s);
        }
        expect(error).toThrow(Error('Format Error'))
    });
});
